freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/dns/DnsRecordService.java

```diff
@@ -36,6 +37,7 @@
 
     public void deleteDnsRecordByFqdn(String environmentCrn, String accountId,  List<String> fqdns) throws FreeIpaClientException {
         Stack stack = stackService.getByEnvironmentCrnAndAccountId(environmentCrn, accountId);
+        MDCBuilder.buildMdcContext(stack);
         FreeIpa freeIpa = freeIpaService.findByStack(stack);
         FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientByAccountAndEnvironment(environmentCrn, accountId);
```

Comment:
we already have the Stack so we can avoid a redundant read by calling `getFreeIpaClientForStack` instead of `getFreeIpaClientByAccountAndEnvironment`